Fast switch code review changes.

AkamaiDASH opened the pull request.

AkamaiDASH commented:
@bbcrddave @LloydW93 @wilaw  and anyone else interested,   I am thinking about being a little bold and committing this into dev very soon.  I have updated the code based on comments.   I need to add just a few more little things:

* Add Some JSDOcs for new events and API
*  Check to see if this is enabled.  If not enabled then I need to switch back stable buffer target to 12 not 20 seconds.  
* We need to decide on a value for buffer target when this is enabled.  Right now I have it at 20 seconds increased from 12.  We could think about 30 seconds as well.
* Last thing I want to do is not block replacement if our buffer level is too low or if we have abandoned recently.   

I should have all of this in by today and Ill leave it open for comments but would like to merge by end of week if possible.

LloydW93 commented on src/streaming/MediaPlayer.js:
```diff
@@ -1958,6 +1970,7 @@ function MediaPlayer() {
         getSelectionModeForInitialTrack: getSelectionModeForInitialTrack,
         getAutoSwitchQuality: getAutoSwitchQuality,
         setAutoSwitchQuality: setAutoSwitchQuality,
+        setFastABRSwitch: setFastABRSwitch,
```
Is a getter counterpart worthwhile here?

AkamaiDASH replied:
I will add one today.

LloydW93 commented on src/streaming/MediaPlayerEvents.js:
```diff
@@ -105,6 +105,10 @@ class MediaPlayerEvents extends EventsBase {
          */
         this.PERIOD_SWITCH_STARTED = 'streamswitchstarted';
+
+        this.QUALITY_CHANGE_START = 'qualityChangeStart';
```
Are these events external? The convention for events that go to the client is to be lower case - it came in in 2.0 iirc.

AkamaiDASH replied:
Yes they are all public facing.  All events in MediaPlayerEvents are public.  Any event in CoreEvent or Protection marked with "public_" will be moved into this class for public access via MediaPlayerEvents,  and all events, core, public, protection etc... are aggregated into Events.js for internal consistent access .   Not sure I follow the comment that events in the client are lower case?

AkamaiDASH replied:
> The convention for events that go to the client is to be lower case - it came in in 2.0 iirc.

@LloydW93 Will Law pointed out the string names of the events.  Some are all lower case and some camel cased.  Is this what you are referring to?   If so this is just and oversight and does not signify external or internal client.  See more info in this new issue https://github.com/Dash-Industry-Forum/dash.js/issues/1435

LloydW93 replied:
Yep - that's what I meant. Sorry for the confusion!

LloydW93 commented:
Other than my (rather nitpicky) comments, LGTM.

AkamaiDASH commented:
OK some updates.  I have disabled this by default for this release.  But I have enabled it in Akamai's player and think we should in Dash IF player as well.  I feel good about this feature I added some new conditions to make it more robust against stalling.   JSDocs are all in place and added a getter for fast switch per @LloydW93 comment.  I have tested quite a bit at this point and feel confident but will ask others to test in their environments.
